src/main/java/org/opentripplanner/index/model/PatternShort.java

```diff
@@ -6,21 +6,48 @@
 import org.opentripplanner.routing.edgetype.TripPattern;
 
 import com.beust.jcommander.internal.Lists;
+import org.opentripplanner.util.model.EncodedPolylineBean;
 
 public class PatternShort {
 
     public String id;
     public String desc;
-    
+    public EncodedPolylineBean geometry;
+
+    /**
+     * Constructor for maintaining backwards compatibility
+     */
     public PatternShort (TripPattern pattern) {
         id = pattern.code;
         desc = pattern.name;
+        geometry = null;
+    }
+
+    /**
+     * Constructor for when geometry is passed
+     */
+    public PatternShort (TripPattern pattern, EncodedPolylineBean geometries) {
+        id = pattern.code;
+        desc = pattern.name;
+        geometry = geometries;
+    }
+
+    /**
+     * If geometry is passed, pattern input must also be List to allow indexing.
+     */
+    public static List<PatternShort> list (List<TripPattern> in, List<EncodedPolylineBean> geometries) {
+        List<PatternShort> out = Lists.newArrayList();
+        for (int i = 0; i < in.size(); i++) out.add(new PatternShort(in.get(i), geometries.get(i)));
```

Comment:
To allow for a null geometries argument, you could do the following:

```suggestion
        for (int i = 0; i < in.size(); i++) {
            out.add(new PatternShort(in.get(i), geometries == null ? null : geometries.get(i)));
        }
```

Reply:
Good catch thank you! Refactored in 1360e72e44678135324c29f7ac4942514fa05270